### PR TITLE
Add support for sentry.io tracing

### PR DESCRIPTION
**Pull request by Half-Shot**
This is part of a plan to start routing errors to Sentry for review rather than trying to grep through the logs after an error has been reported to us.

### CODE REVIEW

File: src/bridge/IrcBridge.ts [hunk 1]
```diff
@@ -523,6 +523,7 @@ export class IrcBridge {
             logMessage(req, "FAILED");
             stats.request(isFromIrc, "fail", req.getDuration());
             this.logMetric(req, "fail");
+            BridgeRequest.HandleExceptionForSentry(req, "failed");
```
Comment (benparsons) [inline src/bridge/IrcBridge.ts]: this should be `fail`, to match the others? FAILED, fail, fail, failed

File: src/bridge/IrcBridge.ts [hunk 2]
```diff
@@ -536,6 +537,7 @@ export class IrcBridge {
             const isFromIrc = Boolean((req.getData() || {}).isFromIrc);
             stats.request(isFromIrc, "fail", req.getDuration());
             this.logMetric(req, "fail");
+            BridgeRequest.HandleExceptionForSentry(req, "dead");
```
Comment (benparsons) [inline src/bridge/IrcBridge.ts]: fail, fail, dead?

File: src/bridge/IrcBridge.ts (hunk 1 above)
Comment (Half-Shot) [reply src/bridge/IrcBridge.ts]: Hmm, tricky. From an english standpoint, I liked state="failed" more than state="fail". However, it does introduce some inconsistency.

Comment (benparsons) [reply src/bridge/IrcBridge.ts]: I don't mind which, just think they should be consistent

**Pull request merged**